pkg/instance/kubernetes.go

```diff
@@ -210,17 +213,8 @@ func (ks kubernetesService) restart(instance *model.Instance, typeSelector strin
 		return fmt.Errorf("multiple deployments found using the selector: %q", selector)
 	}
 
-	name := items[0].Name
-
-	// Scale down
-	prevReplicas, err := scale(deployments, name, 0)
-	if err != nil {
-		return err
-	}
-
-	// Scale up
-	_, err = scale(deployments, name, prevReplicas)
-
+	data := fmt.Sprintf(`{"spec": {"template": {"metadata": {"annotations": {"kubectl.kubernetes.io/restartedAt": "%s"}}}}}`, time.Now().Format(time.RFC3339))
+	_, err = deployments.Patch(context.TODO(), items[0].Name, types.StrategicMergePatchType, []byte(data), metav1.PatchOptions{})
 	return err
```

Comment:
I think we should return our own error. Maybe

`fmt.Errorf("error restarting %q: %v", items[0].Name, err)`

like you said its sometimes hard to find the source of the error. I think one cause of this is that we miss creating our own errors when getting errors from the stdlib or 3rd party libs.